rocks/RocksDatabase.java

```diff
@@ -143,18 +143,27 @@ RocksSession createAndOpenSession(Arguments.Session.Type type, Options.Session o
         return session;
     }
 
-    synchronized Cache cache() {
+    synchronized Cache borrowCache() {
         if (cache == null) cache = new Cache(this);
```

Comment:
we're using `synchronized` to prevent race conditions in this if block. Should we use an `AtomicReference` instead?